cs130_back/handlers/user.go

```diff
@@ -265,6 +265,53 @@ func GetUser(db *gorm.DB, w http.ResponseWriter, r *http.Request) {
 	respondWithJSON(w, http.StatusOK, cr)
 }
 
+// GetUserGroups retrieves and returns the user's group objects
+func GetUserGroups(db *gorm.DB, w http.ResponseWriter, r *http.Request) {
+	vars := r.URL.Query()
+	id, ok := strconv.Atoi(vars["u_id"][0])
+	if ok != nil {
+		respondWithError(w, http.StatusBadRequest, "Invalid user id")
+		return
+	}
+	p := models.User{ID: id}
+	if GetUserByID(db, &p, w) == 0 {
+		return
+	}
+
+	var  groups []models.Group
+
+	p.GetGroups(db, &groups)
+
+	var responses []CreateGroupResponse
+
+	for _, g := range groups{
+		var gr CreateGroupResponse
+		populateGroupResponse(&g, &gr)
+
+		if err := g.GetMembers(db, &gr.Members); err != nil {
+			switch err {
+			default:
+				respondWithError(w, http.StatusInternalServerError, err.Error())
+			}
+		}
+
+		if err := g.GetMeetingTime(db, &gr.MeetingTime); err != nil {
+			switch err {
+			default:
+				respondWithError(w, http.StatusInternalServerError, err.Error())
+			}
+		}
+
```

Comment:
Also, for each group we need to get their requests. Any user of the group can see these requests. 

So, Group needs a model function similar to User's GetInvitations that uses the following query instead. 
```
db.Raw("SELECT * FROM invitations WHERE " + strconv.Itoa(u.GroupID) + " = group_id" + " AND " + "true" + " = type" )
```